sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/ExpressionDescription.java

```diff
@@ -49,17 +49,21 @@
  *     > SELECT ...;
  *      ...
  *
- * `note()` contains some notes for the expression optionally.
+ * `note()` contains some notes for the expression optionally. This property should have 4
+ * leading spaces and end with a newline.
  *
  * `since()` contains version information for the expression. Version is specified by,
- * for example, "2.2.0".
+ * for example, "2.2.0". It should not start with non-number characters.
+ *
+ * `deprecated()` contains deprecation information for the expression optionally, for example,
+ * "Deprecated since 2.2.0. Use something else instead". This property should have 4
```

Comment:
not related to this PR, but why do we have such a weird requirement for the text?

Reply:
To make it pretty in SQL documentation. Actually, we can loose the requirement by adding some logics to handle in `gen-sql-markdown.py`; however, I thought it's better to leave it strict to have uniformed code style.